File: FIL.Web.Admin/ClientApp/models/CreateEventV1/EventScheduleViewModel.ts
import { OccuranceType } from '../../Enums/OccuranceType';
import { EventFrequencyType } from '../../Enums/EventFrequencyType';

export class EventScheduleModel {
  eventId: number;
  startDateTime: string;
  endDateTime: string;
  localStartTime: string;
  localEndTime: string;
  timeZoneOffset: string;
  timeZoneText: string;
  timeZoneAbbrivation: string;
  eventDetailId: number;
  venueId: number;
  dayId: number;
  localStartDateTime: string;
  localEndDateTime: string;
  isEnabled: boolean;
  isCreate: boolean;
  eventFrequencyType: EventFrequencyType;
}

export class EventScheduleViewModel {
  eventScheduleModel: EventScheduleModel;
  success: boolean;
  isValidLink?: boolean;
  isDraft?: boolean;
  currentStep?: number;
  completedStep?: string;
}

export class EventRecurranceInputViewModel {
  eventId: number;
  startDateTime: string;
  endDateTime: string;
  localStartTime: string;
  localEndTime: string;
  timeZoneOffset: string;
  timeZoneText: string;
  timeZoneAbbrivation: string;
  success: boolean;
  isValidLink?: boolean;
  isDraft?: boolean;
  currentStep?: number;
  completedStep?: string;
  eventScheduleId: number;
  scheduleDetailId: number;
  occuranceType: OccuranceType;
  eventFrequencyType: EventFrequencyType;
  localStartDateTime?: string;
  localEndDateTime?: string;
  dayIds: string;
  occuranceCount: number;
}

export class EventRecurranceScheduleModel {
  eventScheduleId: number;
  scheduleDetailId: number;
  dayIds: string;
  startDateTime: string;
  endDateTime: string;
  eventScheduleStartDateTime: string;
  eventScheduleEndDateTime: string;
  localStartDateTime?: string;
  localEndDateTime?: string;
  localEventScheduleStartDateTimeString: string;
  localEventScheduleEndDateTimeString: string;
  localStartTime: string;
  localEndTime: string;
  localStartDateString: string;
  localEndDateString: string;
  isEnabled: boolean;
}

export class EventRecurranceResponseViewModel {
  eventRecurranceScheduleModel: EventRecurranceScheduleModel[];
  success: boolean;
  isValidLink?: boolean;
  isDraft?: boolean;
  currentStep?: number;
  completedStep?: string;
}

File: FIL.Web.Admin/ViewModels/CreateEventV1/EventScheduleViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace FIL.Web.Admin.ViewModels.CreateEventV1
{
  public class EventScheduleViewModel
  {
    public bool Success { get; set; }
    public bool IsValidLink { get; set; }
    public bool IsDraft { get; set; }
    public string CompletedStep { get; set; }
    public short CurrentStep { get; set; }
    public FIL.Contracts.Models.CreateEventV1.EventScheduleModel EventScheduleModel { get; set; }
  }

  public class EventRecurranceViewModel
  {
    [Required]
    public long EventId { get; set; }
    [Required]
    public long EventDetailId { get; set; }
    public bool Success { get; set; }
    public bool IsValidLink { get; set; }
    public bool IsDraft { get; set; }
    public string CompletedStep { get; set; }
    public short CurrentStep { get; set; }
    public long EventScheduleId { get; set; }
    public long ScheduleDetailId { get; set; }
    public DateTime StartDateTime { get; set; }
    public DateTime EndDateTime { get; set; }
    public FIL.Contracts.Enums.OccuranceType OccuranceType { get; set; }
    public FIL.Contracts.Enums.EventFrequencyType EventFrequencyType { get; set; }
    public string DayIds { get; set; }
    public string LocalStartTime { get; set; }
    public string LocalEndTime { get; set; }
    public int OccuranceCount { get; set; }
    public string TimeZoneAbbrivation { get; set; }
    public string TimeZoneOffset { get; set; }
  }

  public class EventRecurranceResponseViewModel
  {
    public bool Success { get; set; }
    public bool IsValidLink { get; set; }
    public bool IsDraft { get; set; }
    public List<FIL.Contracts.Models.CreateEventV1.EventRecurranceScheduleModel> EventRecurranceScheduleModel { get; set; }
  }
}
